src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,127 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int keyHash = hash(key);
+        int bucketNumber = getIndex(keyHash);
+        if (size == 0 || table[bucketNumber] == null) {
+            ifFullResize(table);
+            table[bucketNumber] = new Node<>(keyHash, key, value, null);
+            size++;
+            return;
+        }
+        addToBucketNextNode(key, value, keyHash, bucketNumber);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int position = getIndex(hash(key));
+        Node node = table[position];
+        if (node == null) {
+            return null;
+        }
+        if (node.key == null) {
+            return (V) node.value;
+        } else {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return (V) node.value;
+                }
+                node = node.next;
+            }
+            return null;
+        }
+    }
+
+    private int getIndex(int keyHash) {
+        return keyHash % table.length;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
```

Comment:
do we really need to store hash separately in this class? I think key comparison is enough to find it and we do not use it anywhere